File: data_structures/linkedList.py
class Node:
    def __init__(self, data, ):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self, head):
        self.head = head

    def insertNode(self, newNode):
        currentNode = self.head
        while currentNode.next != None:
            currentNode = currentNode.next

        currentNode.next = newNode

    def deleteNode(self, query):
        currentNode = self.head
        previousNode = None

        while currentNode is not None:
            if currentNode.data == query:
                previousNode.next = currentNode.next
                return

            previousNode = currentNode
            currentNode = currentNode.next

        print('Could not find "%s" in queue' % query)

    def displayData(self):
        currentItem = self.head

        while currentItem is not None:
            print(currentItem.data)
            currentItem = currentItem.next


def main():
    node1 = Node('node 1')
    queue = LinkedList(node1)

    node2 = Node('node 2')
    queue.insertNode(node2)

    node3 = Node('node 3')
    queue.insertNode(node3)

    queue.displayData()

    queue.deleteNode('node 2')
    queue.displayData()

    queue.deleteNode('test')


main()

File: data_structures/linkedList.php
<?php
class Node
{
    public string $data;
    public ?Node $next = null;

    public function __construct($data)
    {
        $this->data = $data;
    }
}

class LinkedList
{
    private Node $head;
    private int $length = 0;

    public function insert(Node $newNode)
    {
        if ($this->length === 0) {
            $this->head = $newNode;
            $this->length++;
            return;
        }

        $currentNode = $this->head;
        while (!empty($currentNode->next)) {
            $currentNode = $currentNode->next;
        }

        $currentNode->next = $newNode;
        $this->length++;
    }

    public function remove(string $query)
    {
        if ($this->length < 1) {
            return;
        }

        $previousNode = null;
        $currentNode = $this->head;

        for ($i = 0; $i < $this->length; $i++) {
            $previousNode = $currentNode;

            if ($currentNode->data === $query) {
                $previousNode->next = $currentNode->next;
                $this->length--;
                unset($currentNode);

                return;
            }

            $currentNode = $currentNode->next;
        }

        echo ("$query not found in list.\n");
    }

    public function display()
    {
        $currentNode = $this->head;
        for ($i = 0; $i < $this->length; $i++) {
            echo ($currentNode->data . "\n");
            $currentNode = $currentNode->next;
        }

        echo ("\n");
    }
}

function main()
{
    $list = new LinkedList();

    $node1 = new Node("Node One");
    $list->insert($node1);

    $node2 = new Node("Node Two");
    $list->insert($node2);

    $node2 = new Node("Node Three");
    $list->insert($node2);
    $list->display();

    $list->remove("Node Two");
    $list->display();

    $list->remove("Node five");
}

main();
